### PR TITLE
Hibernate-5-ify ProviderTypeSetting

### LINKED ISSUE #146: Dentist or physician requires a specialty, but no specialties appear
When enrolling as a physician or as a dentist, the application requires a specialty type to be entered:

specialty-screenshot

But no specialty types appear in the dropdown. We have specialties in the seed data (in table `specialty_types`), so I'm not sure why this is.

[Copying and adding to the issue that @cecilia-donnelly originally filed at NASA-Tournament-Lab/coeci-cms-mpsp#139 .]

### LINKED ISSUE #177: Personal Care Assistant Enrollment Application has no issuing states
As I tried to enroll as a provider in the type "Personal Care Assistant" I was stymied at step 2 (Training) while trying to add a license/certification, because the dropdown list of issuing states was empty.

Could be related to or part of #166.

### PR DESCRIPTION
**Pull request by jasonaowen**
`ProviderTypeSetting` represents a polymorphic, application-side many-to-many relationship between `ProviderType` and several other classes. It was problematic early in the conversion to PostgreSQL because the column storing the application-side foreign key value is textual and some of the relationships were to classes that had numeric primary keys; Oracle seems to have allowed such mistyped comparisons, but PostgreSQL does not, and so those links were extracted in PRs 
#86 and #99.

I thought that there might be remaining such links that hadn't yet caused problems due to being blocked by other incomplete functionality, but after reviewing the seed data and code that uses this class, it turns out all the remaining relationships are to other `LookupEntity`s, which have a textual primary key.

We should, as part of #57, finish extracting these links, but that is tech debt rather than broken functionality, so for now convert this class and mark it deprecated.

Issue #36 Use Hibernate 5, instead of 4
Resolves #146 Dentist or physician requires a specialty, but no specialties appear
Resolves #177 Personal Care Assistant Enrollment Application has no issuing states

### CODE REVIEW

File: psm-app/db/seed.sql [hunk 1]
```diff
@@ -1201,3 +1201,156 @@ CREATE TABLE pay_to_providers(
   npi TEXT,
   phone TEXT
 );
+
+CREATE TABLE provider_type_settings (
```
Comment (cecilia-donnelly) [inline psm-app/db/seed.sql]: I think this needs to be added to the `DROP` list above, too.

File: psm-app/services/src/main/java/gov/medicaid/entities/ProviderTypeSetting.java [hunk 2]
```diff
@@ -15,16 +15,63 @@
...
     private String relatedEntityType;
 
+    /**
+     * The primary key value of the related entity. The related entities always
+     * inherit from LookupEntity, which defines the primary key as a string
+     * named <code>code</code>.
+     *
+     * @see LookupEntity
+     */
+    @Column(name = "related_entity_code")
```
Comment (cecilia-donnelly) [inline psm-app/services/src/main/java/gov/medicaid/entities/ProviderTypeSetting.java]: Thanks for these comments!

File: psm-app/db/seed.sql (hunk 1 above)
Comment (jasonaowen) [inline psm-app/db/seed.sql]: Oops, good catch! Thank you.

**Comment by jasonaowen** [comment]
Amended to add `provider_type_settings` to the `DROP` list. Thanks, @cecilia-donnelly !

**Pull request merged**